src/components/Calculator.jsx

```diff
@@ -1,39 +1,49 @@
-import React from 'react';
+import React, { useState } from 'react';
+import calculate from '../logic/calculate';
 
 function Calculator() {
+  const [total, setTotal] = useState(null);
+  const [next, setNext] = useState(null);
+  const [operation, setOperation] = useState(null);
+  const handleInput = (value) => {
+    const result = calculate({ total, next, operation }, value);
+    setTotal(result.total);
+    setNext(result.next);
+    setOperation(result.operation);
+  };
   return (
     <div className="calculator-wrapper">
       <div className="app">
         <div className="keypad">
-          <p className="result">10</p>
+          <p className="result">{next || total || 0}</p>
           <div>
-            <button className="numbers" type="button" id="clear">AC</button>
-            <button className="numbers" type="button" id="backspace">+/-</button>
-            <button className="numbers" type="button" name="/">&divide;</button>
-            <button className="yellow-numbers" type="button" name="/">&divide;</button>
+            <button className="numbers" onClick={(e) => handleInput(e.target.textContent)} type="button" id="clear">AC</button>
+            <button className="numbers" onClick={(e) => handleInput(e.target.textContent)} type="button" id="backspace">+/-</button>
+            <button className="numbers" onClick={(e) => handleInput(e.target.textContent)} type="button" name="/">&divide;</button>
+            <button className="yellow-numbers" onClick={(e) => handleInput(e.target.textContent)} type="button" name="/">%</button>
```

Comment:
[ Optional Changes ]
This is not a major change but I highly recommend you make the change.

- [ ] I noticed that there is a switch of order for the divide and the modulus button. Kindly rearrange them as it is in the original design. The modulus comes first before the divide sign. Please change the order and switch the background color. 👍🏽